transcom/mymove
# Int b 21668 stored procs for ordering service items

## [B-21668](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1030533)

## Summary

# Testing CreateApprovedServiceItemsForShipment
1. create a new move with one of the addressees going to or from an Alaska location
2. grab the shipment id
3. then go to db client  or psql-dev  and run `CALL public.createapprovedserviceitemsforshipment(:shipment_id); `
4. go to mto_service_items table and see that 
these service items are added to the shipment 
```
International Shipping & Linehaul
International POE Fuel Surcharge
International Origin SIT Fuel Surcharge
International Destination SIT Fuel Surcharge
International HHG pack
International HHG unpack
```

# Testing CreateAccessorialServiceItemsForShipment
1. run `make db_dev_fresh ` or `make db_dev_migrate` to get the db changes into your db client
2. in a new query window set up the call similar to this
```
DO $$
DECLARE
    service_items mto_service_item_type[];
BEGIN
    service_items := ARRAY[
        ROW(
            NULL, -- id
            NULL, -- move_id
            NULL, -- mto_shipment_id
            'f2739142-97d1-40f3-a8f4-6a9daf390806', -- re_service_id
            NOW(), -- created_at
            NOW(), -- updated_at
            'Storage reason', -- reason
            '12345', -- pickup_postal_code
            'Storage service description', -- description
            'SUBMITTED', -- status (valid value from public.service_item_status)
            NULL, -- rejection_reason
            NULL, -- approved_at
            NULL, -- rejected_at
            '54321', -- sit_postal_code
            '2024-01-15', -- sit_entry_date
            '2024-02-01', -- sit_departure_date
            'eb4275bf-7826-412d-a49a-84d923438e53', -- sit_destination_final_address_id
            'eb4275bf-7826-412d-a49a-84d923438e53', -- sit_origin_hhg_original_address_id
            'eb4275bf-7826-412d-a49a-84d923438e53', -- sit_origin_hhg_actual_address_id
            500, -- estimated_weight
            450, -- actual_weight
            'eb4275bf-7826-412d-a49a-84d923438e53', -- sit_destination_original_address_id
            '2024-01-20', -- sit_customer_contacted
            '2024-01-25', -- sit_requested_delivery
            FALSE, -- requested_approvals_requested_status
            TRUE, -- customer_expense
            'Storage fee covered by customer', -- customer_expense_reason
            120, -- sit_delivery_miles
            NULL, -- pricing_estimate
            TRUE, -- standalone_crate
            NULL, -- locked_price_cents
            'O', -- service_location
            null, -- poe_location
            null -- pod_location
        )::mto_service_item_type,
        ROW(
            NULL, -- id
            NULL, -- move_id
            NULL, -- mto_shipment_id
            'ae84d292-f885-4138-86e2-b451855ffbf2', -- re_service_id
            NOW(), -- created_at
            NOW(), -- updated_at
            'Delivery delay reason', -- reason
            '54321', -- pickup_postal_code
            'Delivery service description', -- description
            'APPROVED', -- status (valid value from public.service_item_status)
            NULL, -- rejection_reason
            NULL, -- approved_at
            NULL, -- rejected_at
            '12345', -- sit_postal_code
            '2024-03-01', -- sit_entry_date
            '2024-03-15', -- sit_departure_date
            'eb4275bf-7826-412d-a49a-84d923438e53', -- sit_destination_final_address_id
            'eb4275bf-7826-412d-a49a-84d923438e53', -- sit_origin_hhg_original_address_id
            'eb4275bf-7826-412d-a49a-84d923438e53', -- sit_origin_hhg_actual_address_id
            600, -- estimated_weight
            580, -- actual_weight
            'eb4275bf-7826-412d-a49a-84d923438e53', -- sit_destination_original_address_id
            '2024-03-05', -- sit_customer_contacted
            '2024-03-10', -- sit_requested_delivery
            TRUE, -- requested_approvals_requested_status
            FALSE, -- customer_expense
            NULL, -- customer_expense_reason
            150, -- sit_delivery_miles
            NULL, -- pricing_estimate
            FALSE, -- standalone_crate
            NULL, -- locked_price_cents
            'D', -- service_location
            null, -- poe_location
            null -- pod_location
        )::mto_service_item_type
    ];

3. then  Execute the procedure as below
    ```
CALL CreateAccessorialServiceItemsForShipment(
        ':shipmentId',
        service_items
    );
END $$;
```

### Database

#### Any new migrations/schema changes:
